src/Fantomas.Tests/LetBindingTests.fs

```diff
@@ -1231,7 +1231,7 @@ let x =
                tcref.ILTyconRawMetadata.CustomAttrs.AsArray
                |> Array.exists (fun attr ->
                    attr.Method.DeclaringType.TypeSpec.Name = typeof<TypeProviderEditorHideMethodsAttribute>
-                       .FullName)
+                                                                 .FullName)
```

Comment:
this change is introducing vanity alignment here :-/

Reply:
This is unavoidable when you want to respect the `noBreakLine` infix operators.
[Extreme example](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAKAtgQwC7QAswAPMAdwEtcCAdAJxoDswAfMAZVk0TAH0wAtAD4wARnpNWYAHKxGPYWABMASgjqJzddp3qAvJt1HjJ07joBPQ6Zu3TAZmt3nNytTCJSigCyQANOBQAM6IuLgUjADmQahgANqBEKC6kAAkuBYADshokBG4-onqaQBu2AA2AK458UXaAAwBxpAAkoxIjLjsFABeyE1G3nVgALpFAL4DScNpGdmxeZ2FKallVTUJxqJTOpAAstjEADIRiEeIUdTLRqJK9cNj2pNFybvpWTWQF-AA8gBmJ3k7Ay5X6M1WFWqsU2RiUO20kAAou1-oCwc1yn9IA8JjtXgj3vNcgAjWCwcrXAlrKFoGG6ezw4ptEJ0XAAMQiFWkiDI5VOlJ05mqOKeePBc0%2BpPJAogpUhG2GEG8jNl7EQmAocHKcgAgrhkb8AfyVWA-hUQiL1M9tPjioTJWSKYy5etoYqwABWE2QdiZbDQRAAIUQf1gdEQAAVsHRsNxcIg6DL1ELEJaINb1LbZfaFlKneDqQrjAA2b2%2B-1BkNhs6wMgJ6DYEJtEqwBvhORJiAptNgDPTFYS3OOpMumm1YwAdjLfoDwdD4YAqplsnQG03GC22xQOyazeULbpHlaxQOPkPpc6Ia7ae6ABzTitz6sAYXKjaCz7kQSF0FwYc7prmqmh64i84pniSw6XoWbrGAAnA%2Bs5VuGezqsSCYAXuB46Ee6Ynm8g6QReBbyrBNyNO6PozpW86IJ%2B2qMJhQE9n2YBZlAOZEfmKwwTeWzbJR5YBjqfzxnQn6YDgAHdiBopgaeRJQHmI5XmOdI6LciE0dWaoalqO7ulhwE4aBNrgYpkDKdBpF8TcDKCdRIlibpmrkgZxgySZclmQpDrETxNnjjcyqUW0HS4D8jAACqWAA6lQBBMfuxnaLhvb4QShFKVBJHXkFuiiF6DkVjqdCwJU7QACKIHyGpidJdDCrJx7yQREFQPkKm8flGmlpRBzEC0fxRQQFyIsl7AEGGuDxfAVwmt49zNXhrWZe1iwFNZeXqdoohTv1hxtH8FDED8K54GGiLEJk4ZBEE26MSaHpLV5LU%2BW1FmdVtanuqI94HcQABKiBwHQ8CzfN7qLSxGV2llG1dYFO3qKICEA8DoPwNIlSYOhdD-C08aYDEJqiDDq1w%2BtmCVOU4R8vIbJhjgYQJlFZ7fUWsIUc0GNhvAew03TpyM3QzP1d6hDRv6Ym8JQc2JeT71rZ9Swc2RuhKAJzQDaVMYWD8dAnN%2BEOJQtL2paZmbmZ8X25T9xhKHCAO69g%2BuGxQ37Y7jCYE0TJO-YrVu%2BQs1O0xQ9OICLYus%2Bzducxr9nNC7btG7gAthxHUd4OLlGSzGv4JrLFDy9iy3pRT2bw7bAXbe6SghdrhwAGryoGETwBEkQmwBt7m%2BoaWsexszrdXbzdcjEBKEVjfEGyFW-g9bftJ33dm4H-YfTbqtx%2BrOhKH1M%2BVbAuAAOKhFdN2IHdD2r%2B6z3r2x1vntxY9I3X%2B3a4L4enIG2rQAA1oGfO-9QhBEiuwWMdFyQ40eoZZiZdB5Py4ojWuDt-rNG5LyH%2BoQ6wXDZtkaqx1GBUAejqdoqFvZ0H9sYIyD8h6cWyv5V%2BqDYRo2aAAaUQIgTIQ0RoXDaBA7gaIkrYQtt5IOm8FijypG-Yw9huZGH2IccaGoggEFvsYaGCDYaV3WlZHetl6Ra0Ud0KIoJSqRBxhcGaiBiR7AQOiIwtDtEVw4vDfRNd7ZGHsE7JOfJIiMDnowBechugBLwJUcMUVYBhWsXgB6IiUr90thvZWfkX4yJYfSROiidS0wTIweJJQaxRAoXjQhEQSFfkSXQpBjCMl2nHu6ewDdFHp3CGiI4sZiTwGwK%2BWA90oiYIjjUlxStKYWQ8cwrx9Jp6KMqh7bAxJQQqI9gQdgFhOiHFGa9Fa4zdGTJyp4%2BOOh7AH0UZw7hsTOhtCAdgYJps4HJVqcHZBatDGnI-oo3%2B9z-5oiCKVcq7RuTfkQPzL%2BEcL63XutUk0nkxFvQkWk5%2BKCZmnPQd86MT5wwLKCNAOgFANSFPjPABcxC5DVWgG%2BGM7ZYE0Pgbs8u%2By3F6KOdMk52h7BsJMeYTUacHE7IRemQIIwBh5CCGye6sRaHjCAA).

`=` as an infix operator has very strict rules to what it considers valid.
I don't believe there is anything else valid that having `leftExpr = rightExpr`, `=` needs to after `leftExpr` and `rightExpr` needs to on the righthandside of `=`.

Reply:
mmm, interesting...; but cannot the operator itself (and the rest) be put in the next line?

Reply:
No, this is the behaviour for the other infix operators, but there are a few operators that don't allow this.
These are bundled in https://github.com/fsprojects/fantomas/blob/219bd89abcce8a7dfd6717e7744b78a9c42dc456/src/Fantomas/SourceParser.fs#L19

Reply:
I just tested an F# snippet placing a `=` on the next line, and compilation worked. So I guess `=` shouldn't be in that `noBreak` list?

Reply:
One snippet is most likely a lucky shot, to be honest.
You can try and remove `=` from `noBreakInfixOps`, running the unit tests about 8 of them don't produce valid code anymore. I'm also not trying to solve that particular issue here.
The output of #1863 is confusing and that part, I'm trying to address here.

Reply:
I understand the feeling about the separation of concerns, however, the way I see it, you shouldn't commit a bugfix that introduces a new bug (the bug was not there before).

Reply:
I'm pretty ok with replacing this with a new "bug", it really is the lesser evil. Believe it or not, very few people bring this problem up in the style guide or here. Nor did the style truly embrace that all vanity alignment should be expelled. ([Comment](https://github.com/dotnet/docs/issues/26507#issuecomment-944167218)).

So, yes it can lead to a vanity alignment, which is the safest way to deal with this. [Another example](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWRXc%2B9-8DBQqmAC8YAC4AnAJ5gARmADuiCQAswsAB5gAtAD4wzSCAC%2BQA).
People get way more annoyed with Fantomas breaking their code than with having a vanity alignment.

If users don't like this, maybe consider introducing a local value to capture the multiline expression or increase `fsharp_max_infix_operator_expression`.